### PR TITLE
Send the ad view time to the server

### PR DESCRIPTION
**Pull request by davidfischer**
The view time is counted in 1 second increments up to a 5 min max. View time is only counted if the ad is in the viewport, the tab is the primary one, and the browser is not minimized. Uses the [Page Visibility API](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API) ([supported on IE11](https://caniuse.com/mdn-api_document_visibilitystate))

- Makes an extra request to the ad server when the tab/browser is minimized or about to be closed to send the time the ad was viewed. Any time spent viewing the ad after re-maximizing the browser isn't counted.
- Uses the same "ad view" API but with an extra parameter (**Note**: changes to the ad server will be required to handle this)
- After this change, ad views (even without view time) will require the browser is not minimized and the tab is the primary

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -309,9 +321,50 @@ export class Placement {
...
+          placement.view_time += VIEW_TIME_INTERVAL;
+
+          if (placement.view_time >= VIEW_TIME_MAX) {
+            clearInterval(view_time_counter);
+          }
+        }
+      }, VIEW_TIME_INTERVAL * 1000, placement.target);
+
+      document.addEventListener("visibilitychange", () => {
+        // Check if the tab has gone out of focus or the browser/app is minimized
```
Comment (ericholscher) [inline index.js:339]: Does this also trigger on closing of the tab?

**Review by ericholscher — APPROVED** [review]
Looks like a simple enough change. I assume we're planning to roll this out to the beta client & RTD first, so we can figure out the dynamics of it?

File: index.js (hunk 1 above)
Comment (davidfischer) [reply index.js:339]: Yes. It is supposed to and it did successfully in Chrome as far as I could test in the debugger.

Comment (davidfischer) [reply index.js:339]: I wouldn't mind running this whole thing through an IE11 test as well.

Comment (ericholscher) [reply index.js:339]: We should note that in the comment then, since I imagine that's the most important time this will be fired.

Comment (davidfischer) [reply index.js]: Done

**Pull request merged**